src/network/core/tcp_connect.cpp

```diff
@@ -24,38 +26,277 @@ static std::vector<TCPConnecter *> _tcp_connecters;
  * Create a new connecter for the given address
  * @param connection_string the address to connect to
  */
-TCPConnecter::TCPConnecter(const std::string &connection_string, uint16 default_port) :
-	connected(false),
-	aborted(false),
-	killed(false),
-	sock(INVALID_SOCKET)
+TCPConnecter::TCPConnecter(const std::string &connection_string, uint16 default_port)
 {
-	this->address = ParseConnectionString(connection_string, default_port);
+	NetworkAddress address = ParseConnectionString(connection_string, default_port);
+	this->connection_string = address.GetAddressAsString(false);
```

Comment:
This will actually resolve the name and then convert it back to a string, right? So we need a variant of "ParseConnectionString" that returns it with a valid port number in it, i.e. one that injects the default port number.
A `std::string NormalizeConnectionString(const std::string &connection_string, uint16 default_port)` which is that just calls ParseFullConnectionString and then returns the concatenated returned ip and port number.